New tools implementation

mari-ga opened the pull request:
* Added code for summary python - still not working with mudata
* Added code for both modes: multiple sample and single sample
* Added creation of Seurat object for raw data, no pre-processing for BFF and Demuxmix
* BFF - check under compatible environment 
* Demuxmix and GMM-Demux running on single mode

wxicu commented:
Thanks for your help.  Some general feedback 
- As already mentioned in the last PR, please turn gmm_param.R into python script or though command line, as it causes runtime error for Conda to install the new environment on the cluster. It takes more than ten minutes...
- Make the script executable via `chmod +x ${script}` and don't need to call it like `Rscript blabla or Python blabla`
- Please modify the conda env file for demuxmix and bff, so as to install e.g. demuxmix though bioconductor. I know it is not compatible on your computer but we need this yml for other users.
- I don't think demuxmix and BFF needs pre-processing, because they don't require any normalization, IMO reading the counts can be done in the demuxmix process itself, which saves a little bit runtime, and makes the structure more clear, e.g. a problem here is that the output of demuxmix preprocessing will always overwrite the one of htodemux, as they have exactly the same output path
- Please fix the conflict issue, thanks a lot

mari-ga commented:
Thank you very much for your feedback
1. gmm_param.py replaced the old R version
2. I deleted all Rscript/Python to execute the scripts
3. I added all the missing tools to BFF.yml.   Demuxmix yml was compatible with my computer and it was already working, let me know if you have any issue
4. Pre-processing for BFF and demuxmix deleted. Now demuxmix has its own piece of code for the Seurat object inside the script and BFF received the 10x object for HTO counts directly (it doesn’t need a Seurat object)
5. The error:` iter < maxIter & abs(deltaLogLik/logLik) > tol) { : missing value where TRUE/FALSE needed Calls: demuxmix -> demuxmix -> dmmFitNaive In addition: Warning message: In dnbinom(hto, mu = mu2, size = theta2) : NaNs 	produced`, is not a script or programming error, but a data and parameter tuning error, some datasets produce the error, try it with gx18 raw data that you sent me, it works well. We would need to find and change the right parameters to run the tool on the other datasets.
6. Conflict issue fixed

Let me know, if there's anything else I can do

wxicu commented:
Cool!! Thank you. Some quick feedback before looking into details
3. Demuxmix is available via [Conda.](https://anaconda.org/bioconda/bioconductor-demuxmix) I think currently the user need to install demuxmix inside the process, but we can directly provide a conda env for it, would be great. 
5. tbh I think I already use the raw gx12? Why did I still have the same error? Really weird.
